### PR TITLE
Add `vue/no-console` rule

### LINKED ISSUE #2155: [New Rule] `vue/no-console`
**Please describe what the rule should do:**

Since Vue ^3.3 now [supports console in the template](https://github.com/vuejs/core/issues/7939), a `vue/no-console` rule is needed to regulate its usage.

`vue/no-console` should work exactly the same as [`no-console`](https://eslint.org/docs/latest/rules/no-console) but in the template.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

With `'vue/no-console': 'error'`, the following code should throw:

```vue
<template>
  <button @click="console.log"></button>
</template>
```

### PR DESCRIPTION
**Pull request by ItMaga**
Closes #2155

### CODE REVIEW

**Review by ota-meshi — CHANGES_REQUESTED** [review]
~~Thank you for this PR.
Could you please add test cases to check if the rule works?~~
Forget this comment.
I saw and commented on the wrong diff.

File: tests/lib/rules/no-console.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+tester.run('no-console', rule, {
+  valid: [
+    {
+      filename: 'test.vue',
+      code: `
+      <template>
+        <button @click="Console.log">button</button>
+      </template>
+      `,
+      options: [{ allow: ['log'] }]
```
Comment (ota-meshi) [inline tests/lib/rules/no-console.js:27]: I don't think this option is needed if the intent was to test when the identifiers don't match.

```suggestion
```

**Review by ota-meshi — CHANGES_REQUESTED** [review]

File: tests/lib/rules/no-console.js (hunk 1 above)
Comment (ItMaga) [reply tests/lib/rules/no-console.js]: Done

File: docs/rules/no-console.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+> Disallow the use of `console` in `<template>`
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
+
+## :book: Rule Details
+
+This rule is the same rule as core [no-console] rule but it applies to the expressions in `<template>`.
+
+</eslint-code-block>
```
Comment (ota-meshi) [inline docs/rules/no-console.md:17]: Remove unnecessary end tag.

```suggestion
```

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!

**Review by FloEdelmann — APPROVED** [review]
Looks good to me too :slightly_smiling_face:

**Pull request merged**